src/App.js

```diff
@@ -1,16 +1,34 @@
-import React from 'react';
+import { useState } from 'react';
 import './App.css';
+import ChatLog from './components/ChatLog';
 import chatMessages from './data/messages.json';
 
 const App = () => {
+  const [messages, setMessages] = useState(chatMessages);
+
+  const totalLikes = messages.reduce((total, entry) => {
+    total += entry.liked;
+    return total;
+  }, 0);
+
+  const setLike = (id) => {
+    setMessages((prev) => {
```

Comment:
👍 Nice use of the callback style of our `setState` function. Since the next state of the messages depends on the previous state of the messages, this is our preferred approach, rather than calculating the new value first, then setting it directly.